src/frontier/frontier_cmd_arcade.c

```diff
@@ -0,0 +1,526 @@
+#include "global.h"
+#include "battle/battle_setup.h"
+#include "frontier/frontier_cmd_arcade.h"
+#include "frontier/overlay_80_022340E8.h"
+#include "frontier/overlay_80_02238034.h"
+#include "frontier/overlay_80_02229EE0.h"
+#include "frontier/overlay_80_02239960.h"
+#include "frontier/overlay_80_0222AB40.h"
+#include "frontier/frontier.h"
+#include "launch_application_data.h"
+#include "unk_02096910.h"
+#include "unk_02034354.h"
+#include "unk_02014DA0.h"
+#include "overlay_42.h"
+#include "library/spl_emitter.h"
+
+static void GameBoardArgs_Set(GAME_BOARD_ARGS *args, ArcadeContext *data);
+static void ov80_02233A1C(void *data);
+static void ov80_02233F1C(FrontierContext *ctx, ArcadeContext *arcadeCtx, u32 a2);
+static void ov80_02233F40(struct SPLEmitter *emitter);
+static BOOL ov80_02234028(FrontierContext *ctx);
+
+BOOL FrtCmd_ArcadeAlloc(FrontierContext *ctx) {
+    u32 spC = FrontierScript_ReadVar(ctx);
+    u32 r4 = FrontierScript_ReadVar(ctx);
+    u32 sp10 = FrontierScript_ReadVar(ctx);
+    u32 r6 = FrontierScript_ReadVar(ctx);
+    u32 r7 = FrontierScript_ReadVar(ctx);
+    u16 *sp14 = FrontierScript_ReadVarPtr(ctx);
+
+    FrontierLaunchParam *param = Frontier_GetLaunchParam(ctx->frontierSystem->unk0);
+
+    Frontier_SetData(ctx->frontierSystem->unk0, BattleArcadeData_Alloc(param->saveData, spC, r4, sp10, r6, r7, sp14));
+
+    return FALSE;
+}
+
+BOOL FrtCmd_ArcadeInit(FrontierContext  *ctx) {
+    u32 arg1 = FrontierScript_ReadVar(ctx);
+    void *arg0 = Frontier_GetData(ctx->frontierSystem->unk0);
+    BattleArcadeData_Init(arg0, arg1);
+    return FALSE;
+}
+
+BOOL FrtCmd_ArcadeFree(FrontierContext *ctx) {
+    BattleArcadeData_Free(Frontier_GetData(ctx->frontierSystem->unk0));
+    return FALSE;
+}
+
+extern OVY_MGR_TEMPLATE gOverlayTemplate_BattleArcadeGameBoard;
+
+BOOL ov80_0223371C(FrontierContext *ctx) {
```

Comment:
you can probably name this one tbh, looks like it launches the battle arcade game board

Reply:
true !